### ISSUE TITLE
Aplicação com baixo throughput

### ISSUE THREAD

**Issue by tkrafael** [fredfmiranda/TesteS3SQS#6]
https://github.com/fredfmiranda/TesteS3SQS/blob/f1af930c33819d4b2b3ead9f20294976f4872caa/App.cs#L72
No SQS pode haver situações onde não há itens na fila, e pode haver situações de burst onde centenas ou milhares de mensagens chegarão de uma só vez.
Em uma situação hipotética onde o banco de dados só é capaz de fazer 200 comandos/seg, o código referenciado na issue só consegue fazer 100 transações/seg devido ao `Task.Delay`
Há uma forma de evitar o Task Delay condicionalmente?

**Comment by fredfmiranda** [comment]
@tkrafael sim, aí seria possível ajustar dinamicamente o atraso (Task.Delay) com base na carga de trabalho. Uma abordagem penso eu seria fazer um Task.Delay apenas quando a fila está vazia, para evitar poluir a fila com verificações constantes quando não há trabalho a ser feito. No entanto, se houver mensagens na fila, você pode prosseguir para o próximo ciclo do loop imediatamente.

static async Task Main(string[] args)
{
    while (true)
    {
        try
        {
            var message = await ReceiveMessageFromQueue();
            if (message != null)
            {
                // Processa a mensagem e outras lógicas.

                // Não adiciona nenhum delay se há mensagens para processar
                continue;
            }
            else
            {
                // Se não há mensagens na fila, adiciona um delay para evitar verificações constantes
                await Task.Delay(1000);
            }
        }
        catch (Exception ex)
        {
            Console.WriteLine($"Um erro ocorreu: {ex.Message}");
        }
    }
}

Nesse caso acima, o Task.Delay só é chamado quando não há mensagens na fila. Isso vai permitir que o código processe mensagens da fila o mais rápido possível quando elas estão disponíveis, mas adicione um atraso para evitar sobrecarga quando a fila estiver vazia.

Além disso, para lidar com o limite do banco de dados de 200 comandos/seg, você pode implementar um sistema de rate limiting. Existem algumas bibliotecas que podem ajudar com isso, ou você pode implementar a sua própria lógica para acompanhar o número de operações realizadas e atrasar ou suspender o processamento conforme necessário acredito eu.
